src/util.js

```diff
@@ -87,12 +90,45 @@ export function serializeForm(form) {
 				values = [values];
 			}
 			values.forEach(value => {
-				let param = [name, value].map(encodeURIComponent).join("=");
-				params.push(param);
+				params.setParameter(name, value);
 			});
 		}
 		return params;
-	}, []).join("&");
+	}, params).serialize();
+}
+
+class FormDataParams {
+	constructor() {
+		this.formData = new FormData();
+	}
+
+	setParameter(name, value) {
```

Comment:
```suggestion
	add(name, value) {
```

(same for `StringParams`, of course)

Reply:
Yes, that seems like a good idea.